studio/components/interfaces/Settings/General/CustomDomainConfig/CustomDomainsConfigureHostname.tsx

```diff
@@ -14,30 +14,47 @@ import {
 } from 'components/ui/Forms'
 import { useCustomDomainCreateMutation } from 'data/custom-domains/custom-domains-create-mutation'
 
-const FORM_ID = 'custom-domains-form'
-
 const schema = yup.object({
   domain: yup.string().required('A value for your custom domain is required'),
 })
 
-export type CustomDomainsConfigureHostnameProps = {
-  projectRef?: string
-}
-
-const CustomDomainsConfigureHostname = ({ projectRef }: CustomDomainsConfigureHostnameProps) => {
+const CustomDomainsConfigureHostname = () => {
   const { ui } = useStore()
+  const { ref } = useParams()
+  const { mutateAsync: createCustomDomain } = useCustomDomainCreateMutation()
+  const { services } = useProjectSettings(ref as string | undefined)
+
+  // Get the API service
+  const API_SERVICE_ID = 1
+  const apiService = services ? services.find((x: any) => x.app.id == API_SERVICE_ID) : {}
+  const endpoint = apiService?.app_config?.endpoint
```

Comment:
Are you able to use the new `useProjectSettingsQuery()` hook here? It'll make this all type safe :)

Here are a couple of usage examples that should be helpful here:

https://github.com/supabase/supabase/blob/09dce53d1a919226bb6d1a3415d2ad377115a3a9/studio/components/interfaces/Settings/General/CustomDomainConfig/CustomDomainConfig.tsx#L23-L25

https://github.com/supabase/supabase/blob/09dce53d1a919226bb6d1a3415d2ad377115a3a9/studio/components/interfaces/Settings/General/CustomDomainConfig/CustomDomainVerify.tsx#L137